README.md

```diff
@@ -1,16 +1,1261 @@
+The invoker signals to the executor that a task associated with a UCAN SHOULD be performed.
+
+The invoker MUST be the UCAN delegator. Their DID MUST be authenticated in the `iss` field of the contained UCAN.
+
+### 2.1.2 Executor
+
+The executor is directed to perform some task described in the UCAN by the invoker.
+
+The executor MUST be the UCAN delegate. Their DID MUST be set the in `aud` field of the contained UCAN.
+
+## 2.2 Components 
+
+![](./diagrams/concepts.svg)
+
+### 2.2.1 Closure
+
+A [Closure](#3-closure) is like a deferred function application: a request to perform some action on a resource with specific inputs.
+
+### 2.2.2 Task
+
+A [Task](#4-task) extends a Closure with additional metadata that is not used to describe the meaning of the computation or effect to be run.
+
+### 2.2.3 Batch
+
+A [Batch](#5-batch) is a way of requesting more than one action at once.
+
+### 2.2.4 Invocation
+
+An [Invocation](#6-invocation) is the cryptographically signed container for a Batch. This is the step that "forces" the "deferred" Closure.
+
+### 2.2.5 Pointers
+
+An [Invocation Pointer](#7-pointer) identifies a specific invocation. An Invoked Task Pointer points to a unique Task inside an Invocation.
+
+### 2.2.6 Result
+
+A [Result](#8-result) is the output of a Closure.
+
+### 2.2.7 Receipt
+
+A [Receipt](#9-receipt) describes the output of an invocation, referenced by its Invocation Pointer.
+
+### 2.2.8 Promise
+
+A [promise](#10-promise) is a reference to the receipt of an action that has yet to return a receipt.
+
+## 2.3 IPLD Schema
+
+``` ipldsch
+type Closure struct {
+  with   URI
+  do     Ability
+  inputs Any
+}
+
+type Task struct {
+  with   URI
+  do     Ability
+  inputs Any
+  meta {String : Any} (implicit {})
+}
+
+type Batch union {
+  | Named {String : Task}
+  | List  [Task]
+}
+
+type Invocation struct {
+  uiv  SemVer
+  run  Batch
+  prf  [&UCAN]
+  nnc  String
+  meta {String : Any} (implicit {})
+  sig  Varsig
+}
+
+type InvocationPointer union {
+  | "/" -- Relative to the current invocation
+  | &TaskInvocation
+}
+
+type InvokedTaskPointer struct {
+  envl  InvPtr
+  label String
+} representation tuple
+
+type Receipt struct {
+  ran  &InvocationPointer
+  out  {String : Result}
+  rec  {String : &Receipt}
+  meta {String : Any}
+  sig  Varsig
+} 
+
+type Result union {
+  | Any ("ok")  -- Success
+  | Any ("err") -- Failure
+} representation keyed
+
+type Promise union {
+  | InvokedTaskPointer "ucan/ok"
+  | InvokedTaskPointer "ucan/err"
+  | InvokedTaskPointer "ucan/ptr"
+} representation keyed
+```
+
+# 3 Closure
+
+A Closure is the smallest unit of work that can be requested from a UCAN. It describes one `(resource, ability, inputs)` triple. The `inputs` field is free form, and depend on the specific resource and ability being interacted with, and not described in this specification.
+
+Using the JavaScript analogy from the introduction, a Closure is similar to wrapping a call in an anonymous function:
```

Comment:
As far as I can tell `Closure` only differs from `Capability` by naming of fields. We've been using `Capability` instead, not suggesting it's better but worth evaluating tradeoffs.

Reply:
In re-reading the LaCasa paper, among others in the E-Rights/Ocap space, I do think closure makes more sense than capability from a terminology level in that a closure with its environment represents some contextual evaluation where capabilities typically exist within an object (or closure, which in the lisp sense is related to an object itself).